src/Plugin/RemoteCheck/RemoteRegister.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Drupal\contentpool_client\Plugin\RemoteCheck;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\relaxed\Entity\RemoteInterface;
+use Drupal\relaxed\Plugin\RemoteCheckBase;
+use GuzzleHttp\Psr7\Uri;
+use GuzzleHttp\RequestOptions;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Serializer\Serializer;
+
+/**
+ * @RemoteCheck(
+ *   id = "remote_register",
+ *   label = "Register on remote",
+ * )
+ */
+Class RemoteRegister extends RemoteCheckBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The serializer.
+   *
+   * @var \Symfony\Component\Serializer\Serializer
+   */
+  protected $serializer;
+
+  /**
+   * RemoteRegister constructor.
+   *
+   * @param array $configuration
+   * @param $plugin_id
+   * @param $plugin_definition
+   * @param \Symfony\Component\Serializer\Serializer $serializer
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, Serializer $serializer) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->serializer = $serializer;
+  }
+
+  /**
+   * @param \Drupal\contentpool_client\Plugin\RemoteCheck\ContainerInterface $container
+   * @param array $configuration
+   * @param $plugin_id
+   * @param $plugin_definition
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('serializer')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute(RemoteInterface $remote) {
+    $url = (string) $remote->uri();
+
+    // As the remote targets the relaxed endpoint we have to parse the url
+    // to get the base host.
+    $url_parts = parse_url($url);
+    $base_url = $url_parts['scheme'] . '://' . $url_parts['user'] . ':' . $url_parts['pass'] . '@' . $url_parts['host'];
+
+    /** @var \GuzzleHttp\Client $client */
+    $client = \Drupal::httpClient();
+
+    try {
+      $response = $client->post($base_url . '/_remote-registration?_format=json', $this->generateRegistrationPayload());
+
+      if ($response->getStatusCode() === 200) {
+        $this->result = TRUE;
+        $this->message = t('Registration on remote is valid.');
+      }
+      else {
+        $this->message = t('Remote returns status code @status.', ['@status' => $response->getStatusCode()]);
+      }
+    }
+    catch (\Exception $e) {
+      $this->message = $e->getMessage();
+      watchdog_exception('relaxed', $e);
+    }
+  }
+
+  /**
+   * Collects the data necessary for registration on remote.
+   */
+  private function generateRegistrationPayload() {
+    // Basic Site information.
+    $config = \Drupal::config('system.site');
```

Comment:
needs to be dependency injected...
you can use service-utils for many dependencies here - please add it.